server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -536,17 +546,25 @@ private static Map<String, Mapper> buildMergedMappers(
             ObjectMapper existing,
             Mapper mergeWith,
             MergeReason reason,
-            MapperBuilderContext objectBuilderContext
+            MapperMergeContext objectMergeContext
         ) {
-            Map<String, Mapper> mergedMappers = null;
+            // need to preserve order to make it deterministic which fields are ignored
+            Map<String, Mapper> mergedMappers = new LinkedHashMap<>(existing.mappers);
```

Comment:
While I think that making it more deterministic which fields get added vs ignored makes sense, I don't think it's a huge issue if it isn't. So if you think this would cause more issues than it solves (such as performance and memory impact), I can just revert this to a regular `HashMap` and use `Map.copyOf` instead of `Collections.unmodifiableMap`. I think the biggest benefit is that it does make the testing a bit easier if you know deterministically, which fields you can expect to be in the mapping vs which ones get ignored.

Reply:
I've removed the new order-preserving semantics and adjusted the tests so that they don't assume ordering.